clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py

```diff
@@ -775,6 +783,8 @@ def write_bids(
             metadata["bids_filename"],
             True,
         )
+        dcm2niix_bids_path.append(bids_full_path)
```

Comment:
Maybe something along those lines to write rows within the loop only if the `dcm2niix` conversion was successful?

```python
if dcm2niix_success:
    row_to_write = metadata.to_csv(
        index=False, header=False, lineterminator="\t",
    ).rstrip("\t")
    with open(scans_filepath, "a") as scans_file:
         scans_file.write(row_to_write)
    if "dwi" in metadata.bids_filename and "Philips" in metadata.manufacturer:
        merge_philips_diffusion(...)
```